File: src/example_laser.cpp
#include "ros/ros.h"
#include <cstdlib>
#include "sensor_msgs/LaserScan.h"


void laserCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
{
  ROS_INFO("%f", msg->ranges[msg->ranges.size()/2]);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "example");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("scan_filtered", 1, laserCallback);
  ros::spin();
  return 0;
}

File: src/examples/example_laser.py
#!/usr/bin/python
import rospy
from time import sleep

from sensor_msgs.msg import LaserScan

node = rospy.init_node('example')

def scan_callback(data):
	distance_front = data.ranges[len(data.ranges)//2]
	print(distance_front)
	
rospy.Subscriber('scan_filtered', LaserScan, scan_callback)

# don't terminate
rospy.spin()
